src/Services/PaymentCreation.php

```diff
@@ -133,6 +134,26 @@ public function createPayment($mopId, ResponseAbstract $response, Basket $basket
             $payment->receivedAt = date('Y-m-d H:i:s');
         }
 
+        try {
+            /** @var CurrencyExchangeRepositoryContract $currencyService */
+            $currencyService = pluginApp(CurrencyExchangeRepositoryContract::class);
+
+            $defaultCurrency = $currencyService->getDefaultCurrency();
+
+            if ($payment->currency != $defaultCurrency) {
+                $payment->exchangeRatio = $currencyService->getExchangeRatioByCurrency($payment->currency);
+                $payment->amount = round(
+                    $currencyService->convertToDefaultCurrency(
+                        $payment->currency,
+                        $payment->amount,
+                        $payment->exchangeRatio
+                    ),
+                    2);
```

Comment:
This looks wrong to me. I think the original amount in the foreign currency needs to be saved in the payment. Am I right @j-konopka ?

Reply:
Yes it is better to save the original amount. If their is a foreign currency you have to set the flag "isSystemCurrency" to "false"

Reply:
@j-konopka @raphael-adam ok. We can save the original amount(the amount in foreign currency) but then the old payment UI is not working properly. Also, take a look at the other payment processors, all of them have the same calculation. This bug fix was added because before this fix, the amount was saved in the system currency. Note that if we want to save the foreign amount, we have to change it for all payment methods and another problem is that the isSystemCurrency set to false will not work.

Reply:
Isn't the behaviour like that? :
The amount of the payment is always saved in the currency of the payment. If the payment currency is not the system currency, the exchange rate is saved and `isSystemCurrency` is set to false.
Maybe we can have a call sometime where @j-konopka can explain how the fields should be set.

Reply:
If the `isSystemCurrency` Flag is `true` the amount should be in the system currency. It could be that the currency sign is different to the system currency. But for this case an exchangeRatio should be available.
If the `isSystemCurrency` Flag is `false` the amount should be in the foreign currency and the currency sign should be also the foreign currency.